File: JavaScript_Courses/JavaScript_Basics/JS_BP_2022_10/04_For_Loop_Exercise/03_Histogram.js
function histogram(input) {
    //     < 200	
    // 200 … 399	
    // 400 … 599	
    // 600 … 799	
    // ≥ 800	
    let p1 = 0;  // count
    let p2 = 0;
    let p3 = 0;
    let p4 = 0;
    let p5 = 0;

    let numbersCount = Number(input[0]);

    for (let i = 0; i < numbersCount; i++) {
        let currentNumber = Number(input[i + 1]);
        if (currentNumber < 200) {
            p1++;
        } else if (currentNumber < 400) {  // 200 И 400  (currentNumber >= 200 && currentNumber < 400)
            p2++;
        } else if (currentNumber < 600) {  // 400 И 600
            p3++;
        } else if (currentNumber < 800) {
            p4++;
        } else if (currentNumber >= 800) {
            p5++;
        }
    }

    console.log((p1 / numbersCount * 100).toFixed(2) + "%");  // (p1 / (p1 + p2 + p3 + p4 + p5))
    console.log((p2 / numbersCount * 100).toFixed(2) + "%");
    console.log((p3 / numbersCount * 100).toFixed(2) + "%");
    console.log((p4 / numbersCount * 100).toFixed(2) + "%");
    console.log((p5 / numbersCount * 100).toFixed(2) + "%");
}


// histogram(["3", "1", "2", "999"]);

// 1. currentNumber = Number(input[1]);
// 2. currentNumber = Number(input[2]);
// 3. currentNumber = Number(input[3]);

File: Python_Courses/Python_Basics/Python_PB_2023_10/04_For_Loop_Exercise/03_Histogram.py
p1_count = 0
p2_count = 0
p3_count = 0
p4_count = 0
p5_count = 0

numbers_count = int(input())

for _ in range(numbers_count):  # [0; numbers_count - 1] => count == numbers_count
    current_number = int(input())
    if current_number < 200:
        p1_count += 1  # p1_count = p1_count + 1
    elif current_number < 400:  # [200; 400) == [200; 399]
        p2_count += 1
    elif current_number < 600:
        p3_count += 1
    elif current_number < 800:
        p4_count += 1
    else:  # eq. elif current_number >= 800:
        p5_count += 1

# percent = target / total * 100
# total_count = p1_count + p2_count + p3_count + p4_count + p5_count

print(f"{p1_count / numbers_count :.2%}")
print(f"{p2_count / numbers_count :.2%}")
print(f"{p3_count / numbers_count :.2%}")
print(f"{p4_count / numbers_count :.2%}")
print(f"{p5_count / numbers_count :.2%}")
